pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/startree/StarTreeV2Metadata.java

```diff
@@ -44,8 +44,12 @@ public StarTreeV2Metadata(Configuration metadataProperties) {
     _functionColumnPairs = new HashSet<>();
     for (String functionColumnPair : metadataProperties.getStringArray(MetadataKey.FUNCTION_COLUMN_PAIRS)) {
       _functionColumnPairs.add(AggregationFunctionColumnPair.fromColumnName(functionColumnPair));
-    }
-    _maxLeafRecords = metadataProperties.getInt(MetadataKey.MAX_LEAF_RECORDS);
+      Configuration functionColPairsConfig =
+          metadataProperties.subset(MetadataKey.AGGREGATION_CONFIG + "." + functionColumnPair);
+      if (!functionColPairsConfig.isEmpty()) {
+        _functionColumnPairs.add(AggregationFunctionColumnPair.fromConfiguration(functionColPairsConfig));
```

Comment:
(CRITICAL) This will cause the same aggregation being added twice if the codec is different

Reply:
Using a set here, I might be wrong but TreeSet can't contain duplicates.

Reply:
I've changed the data structure to a treeMap with functionColumnPair as a string key to avoid duplicates.